### PR TITLE
WIP: feat: add OpenDealPending() func

### PR DESCRIPTION
**Pull request by Sokel**

### CODE REVIEW

File: blockchain/api.go [hunk 1]
```diff
@@ -174,6 +182,46 @@ func (bch *api) OpenDeal(key *ecdsa.PrivateKey, deal *pb.Deal) (*types.Transacti
...
 
+func (bch *api) OpenDealPending(key *ecdsa.PrivateKey, deal *pb.Deal, wait time.Duration) (*big.Int, error) {
+	tx, err := bch.OpenDeal(key, deal)
+	if err != nil {
+		return nil, err
+	}
+
+	fmt.Println(tx.Hash().String())
+
+	ctx, cancel := context.WithTimeout(context.Background(), wait)
```
Comment (antmat) [inline blockchain/api.go]: I'm not sure background context is the best choice

File: blockchain/api.go [hunk 2]
```diff
@@ -147,6 +152,9 @@ var DealOpenedTopic common.Hash = common.HexToHash("0x873cb35202fef184c9f8ee23c0
 var DealAcceptedTopic common.Hash = common.HexToHash("0x3a38edea6028913403c74ce8433c90eca94f4ca074d318d8cb77be5290ba4f15")
 var DealClosedTopic common.Hash = common.HexToHash("0x72615f99a62a6cc2f8452d5c0c9cbc5683995297e1d988f09bb1471d4eefb890")
 
+var ErrorLogNotFound = errors.New("Log not found")
+var ErrorTxFailed = errors.New("transaction failed")
```
Comment (antmat) [inline blockchain/api.go]: I thought we agreed not to make cons errors used in a single context (I, personally, don't mind)

File: blockchain/api.go [hunk 3]
```diff
@@ -174,6 +182,46 @@ func (bch *api) OpenDeal(key *ecdsa.PrivateKey, deal *pb.Deal) (*types.Transacti
...
+	fmt.Println(tx.Hash().String())
+
+	ctx, cancel := context.WithTimeout(context.Background(), wait)
+	defer cancel()
+
+	tk := time.NewTicker(1 * time.Second)
+	defer tk.Stop()
+
+	for {
+		select {
```
Comment (antmat) [inline blockchain/api.go]: We always wait at least 1 second - is it ok?

File: blockchain/api.go [hunk 4]
```diff
@@ -174,6 +182,46 @@ func (bch *api) OpenDeal(key *ecdsa.PrivateKey, deal *pb.Deal) (*types.Transacti
 	return tx, err
 }
 
+func (bch *api) OpenDealPending(key *ecdsa.PrivateKey, deal *pb.Deal, wait time.Duration) (*big.Int, error) {
+	tx, err := bch.OpenDeal(key, deal)
+	if err != nil {
+		return nil, err
+	}
+
+	fmt.Println(tx.Hash().String())
```
Comment (sshaman1101) [inline blockchain/api.go]: Debug with prints is awesome :)

File: blockchain/api.go [hunk 5]
```diff
@@ -174,6 +182,46 @@ func (bch *api) OpenDeal(key *ecdsa.PrivateKey, deal *pb.Deal) (*types.Transacti
...
+			if err != nil {
+				if err == ethereum.NotFound {
+					break
+				}
+				return nil, err
+			}
+			if txReceipt.Status == types.ReceiptStatusSuccessful {
+				for _, l := range txReceipt.Logs {
+					if l.Topics[0] == DealOpenedTopic {
+						return l.Topics[3].Big(), nil
```
Comment (oopcode) [inline blockchain/api.go]: Are those indices guaranteed to exist? I have no idea, just asking.

File: blockchain/api.go [hunk 6]
```diff
@@ -147,6 +152,9 @@ var DealOpenedTopic common.Hash = common.HexToHash("0x873cb35202fef184c9f8ee23c0
 var DealAcceptedTopic common.Hash = common.HexToHash("0x3a38edea6028913403c74ce8433c90eca94f4ca074d318d8cb77be5290ba4f15")
 var DealClosedTopic common.Hash = common.HexToHash("0x72615f99a62a6cc2f8452d5c0c9cbc5683995297e1d988f09bb1471d4eefb890")
 
+var ErrorLogNotFound = errors.New("Log not found")
```
Comment (sshaman1101) [inline blockchain/api.go]: Just a "log"? I think it's too general, so just add the "transaction" word to an error text :)

File: blockchain/api.go (hunk 5 above)
Comment (sshaman1101) [reply blockchain/api.go]: slice indices without length checking...? Looks fragile.

**Pull request merged**